File: VB/SinglePageReport/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

Namespace SinglePageReport

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim list As List(Of Data) = New List(Of Data)()
            Dim random As Random = New Random()
            For i As Integer = 0 To 500 - 1
                Dim data As Data = New Data() With {.Name = String.Format("Name{0}", i), .Group = String.Format("Group{0}", random.Next(10)), .Description = String.Format("Item Description {0}", i), .Value = random.Next(1000)}
                list.Add(data)
            Next

            Dim report As SampleReport = New SampleReport() With {.DataSource = list}
            GenerateSinglePageReport(report)
            Dim printTool As ReportPrintTool = New ReportPrintTool(report)
            printTool.ShowPreviewDialog()
        End Sub
    End Class
End Namespace

File: CS/SinglePageReport/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;

namespace SinglePageReport
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            List<Data> list = new List<Data>();
            Random random = new Random();

            for (int i = 0; i < 500; i++)
            {
                Data data = new Data()
                {
                    Name = string.Format("Name{0}", i),
                    Group = string.Format("Group{0}", random.Next(10)),
                    Description = string.Format("Item Description {0}", i),
                    Value = random.Next(1000)
                };

                list.Add(data);
            }

            SampleReport report = new SampleReport() { DataSource = list };

            SinglePageHelper.GenerateSinglePageReport(report);

            ReportPrintTool printTool = new ReportPrintTool(report);
            printTool.ShowPreviewDialog();
        }
    }
}
